doc/mod/models.rst

```diff
@@ -0,0 +1,107 @@
+Models
+======
+
+.. automodule:: bluetooth_mesh.models
+
+Each model must declare its id and a list of supported opcodes:
+
+.. literalinclude:: examples/model.py
+   :language: python
+
+Sending messages
+----------------
+
+There are two basic methods to send messages to remote nodes:
+:func:`~Model.send_app` and :func:`~Model.send_dev`. Usage should be
+self-explanatory. Keep in mind that they are implemented on top of
+:py:data:`bluetooth_mesh.messages.AccessMessage`, so you don't need to deal with raw
+encoding - just pass the message contents as a :py:data:`dict` aligned with
+message structure. See :py:mod:`bluetooth_mesh.messages` for details.
+
+To handle application retransmissions, use :func:`~Model.repeat`. Basic usage
+looks like this:
+
+.. literalinclude:: examples/repeat.py
+   :language: python
+
+The :func:`~Model.repeat` method takes a callable, because each application
+retransmission might have slightly different content (e.g.  `delay` in
+:py:data:`GenericOnOff` message). This means that you can use it like this:
+
+.. literalinclude:: examples/delay.py
+   :language: python
+
+.. note::
+   It might be more convenient to replace the callable with async generator,
+   but I'm not sure about compatibility with older Python versions.
+
+.. note::
+   Publication API is not implemented yet.
```

Comment:
No longer true (?).

Reply:
We support subscriptions, not publications. This is still missing.

Reply:
Oh, my bad.